### PR TITLE
Trim special characters from cherrypick comments

### PR DESCRIPTION
**Pull request by kargakis**
Fixes https://github.com/kubernetes/test-infra/issues/6371

/cc stevekuznetsov

### CODE REVIEW

**Comment by kargakis** [comment]
@mfojtik ptal

**Comment by stevekuznetsov** [comment]
@dobbymoodge that stupid ellipsis strikes again

File: experiment/cherrypicker/server.go [hunk 1]
```diff
@@ -174,7 +174,7 @@ func (s *Server) handleIssueComment(l *logrus.Entry, ic github.IssueCommentEvent
 	if len(cherryPickMatches) == 0 || len(cherryPickMatches[0]) != 2 {
 		return nil
 	}
-	targetBranch := cherryPickMatches[0][1]
+	targetBranch := strings.TrimSpace(cherryPickMatches[0][1])
```
Comment (stevekuznetsov) [inline experiment/cherrypicker/server.go]: Is it space that is breaking? I thought it was the fact that the last character in that title is an ellipsis and non-ascii?

Comment (kargakis) [reply experiment/cherrypicker/server.go]: Special character (`\r` in this case) which is added by hiting Enter after the comment. See the unit test update. It's what we hit in the [first linked issue](https://github.com/openshift/origin/pull/15847#issuecomment-359410337) and I managed to reproduce in the [second](https://github.com/openshift/openshift-ansible/pull/6806#issuecomment-359500198).

Comment (dobbymoodge) [reply experiment/cherrypicker/server.go]: image

Comment (stevekuznetsov) [reply experiment/cherrypicker/server.go]: OK, have you tested that the ellipsis also doesn't break things?

**Pull request merged**